include/mruby/gc.h

```diff
@@ -21,6 +21,7 @@ struct mrb_state;
 #define MRB_EACH_OBJ_BREAK 1
 typedef int (mrb_each_object_callback)(struct mrb_state *mrb, struct RBasic *obj, void *data);
 void mrb_objspace_each_objects(struct mrb_state *mrb, mrb_each_object_callback *callback, void *data);
+const mrb_int mrb_objspace_page_slot_size();
```

Comment:
This feels like it should be just a #DEFINE, however when using a `const int` return value compilers optimize this function call to a single `MOV` instruction instead, which is okay

https://godbolt.org/z/1b6Prq

Reply:
Since the translation unit is divided, it is usually a function call (I don't know `gcc -flto` :smiley_cat:).
However, I don't think there is any problem with function calls.
If you care about that, you should be fine with global variables (with const).

Also, if you qualify a non-pointer return type with `const`, the compiler will ignore it (`gcc -Wignored-qualifiers`).

Reply:
You're right I keep forgetting about translation unit boundaries.

It feels silly for a function call which always returns the same value. What I mean is because of the way the code is arranged in the c and header file, and because matz prefers not exposing internal structures in the header, it's not possible to create a global const. It's not a big deal either way.